#!/usr/bin/env python
PACKAGE = "hydrus_xi"

from dynamic_reconfigure.parameter_generator_catkin import *

GAIN_FLAG = 0
Z_P_GAIN = 1
Z_I_GAIN = 2
Z_D_GAIN = 3
XY_P_GAIN = 4
XY_I_GAIN = 5
XY_D_GAIN = 6
YAW_P_GAIN = 7
YAW_I_GAIN = 8
YAW_D_GAIN = 9
RP_P_GAIN = 10
RP_I_GAIN = 11
RP_D_GAIN = 12

gen = ParameterGenerator()

gen.add("gain_flag", bool_t, GAIN_FLAG, "Gain Flag", False)
gen.add("z_p", double_t, Z_P_GAIN, "Z P Gain", 5.0, 0, 100)
gen.add("z_i", double_t, Z_I_GAIN, "Z I Gain", 0.3, 0, 10)
gen.add("z_d", double_t, Z_D_GAIN, "Z D Gain", 2.0, 0, 100)
gen.add("xy_p", double_t, XY_P_GAIN, "XY P Gain", 5.0, 0, 100)
gen.add("xy_i", double_t, XY_I_GAIN, "XY I Gain", 0.1, 0, 10)
gen.add("xy_d", double_t, XY_D_GAIN, "XY D Gain", 3.0, 0, 100)
gen.add("yaw_p", double_t, YAW_P_GAIN, "Yaw P Gain", 2.0, 0, 100)
gen.add("yaw_i", double_t, YAW_I_GAIN, "Yaw Yaw I Gain", 0.05, 0, 10)
gen.add("yaw_d", double_t, YAW_D_GAIN, "Yaw D Gain", 4.0, 0, 100)
gen.add("rp_p", double_t, RP_P_GAIN, "Roll Pitch P Gain", 5.0, 0, 100)
gen.add("rp_i", double_t, RP_I_GAIN, "Roll Pitch I Gain", 0.1, 0, 10)
gen.add("rp_d", double_t, RP_D_GAIN, "Roll Pitch D Gain", 10.0, 0, 100)

exit(gen.generate(PACKAGE, "hydrus_xi", "FullyActuatedControllerGains"))
